gobblin-audit/src/main/java/gobblin/audit/policy/column/AbstractColumnProjectionPolicy.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.audit.policy.column;
+
+import gobblin.audit.auditor.ValueAuditRuntimeMetadata;
+
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+
+/**
+ * A base {@link ColumnProjectionPolicy} that reads <code>jobProps</code> to initialize the key and delta columns to project for a table.
+ * <ul>
+ * <li> The field locations to projects are an ordered string specifying the location of the nested field to retrieve.
+ * For example, field1.nestedField1 takes the the value of the field "field1" of the record, and retrieves the field "nestedField1" from it
+ * <li> Key fields/column locations to project are set at {@value #COLUMN_PROJECTION_KEY_FIELD_LOCATIONS} as a comma separated list of field locations.
+ * For each table set {@value #COLUMN_PROJECTION_KEY_FIELD_LOCATIONS}.database.table with the list of columns to project
+ * <li> Delta fields/column locations to project are set at {@value #COLUMN_PROJECTION_DELTA_FIELD_LOCATIONS} as a comma separated list of field locations.
+ * For each table set <code>{@value #COLUMN_PROJECTION_DELTA_FIELD_LOCATIONS}.database.table</code> with the list of columns to project
+ * <li> The default values for projection field locations is empty. Meaning no fields to project.
+ * </ul>
+ */
+public abstract class AbstractColumnProjectionPolicy implements ColumnProjectionPolicy {
+
+  private static final String COLUMN_PROJECTION_KEY_FIELD_LOCATIONS ="gobblin.audit.column.projection.key.field.locations";
+  private static final String COLUMN_PROJECTION_DELTA_FIELD_LOCATIONS = "gobblin.audit.column.projection.delta.field.locations";
+
+  private static final String DOT = ".";
+
+  private final List<String> keyFieldLocations;
+  private final List<String> deltaFieldLocations;
+
+  private static final Splitter SPLITTER_ON_COMMA = Splitter.on(',').trimResults().omitEmptyStrings();
+
+  private AbstractColumnProjectionPolicy(String commaSeparatedKeyfieldLocations,
```

Comment:
I think the key field list and change field list should be part of the metadata object. They should be derived based on the schema.